facade/logstash.go

```diff
@@ -181,51 +201,66 @@ func writeLogstashConfiguration(templates map[string]servicetemplate.ServiceTemp
 	return nil
 }
 
-func getFiltersFromTemplates(services []servicedefinition.ServiceDefinition, filterDefs map[string]string, logFiles *[]string) string {
-	filters := ""
-	for _, svc := range services {
-		filters += getFilters(svc.Name, svc.LogConfigs, filterDefs, logFiles)
-		if len(svc.Services) > 0 {
-			subFilts := getFiltersFromTemplates(svc.Services, filterDefs, logFiles)
-			filters += subFilts
-		}
-	}
-	return filters
-}
 
-func getFilters(serviceName string, configs []servicedefinition.LogConfig, filterDefs map[string]string, logFiles *[]string) string {
-	filters := ""
-	for _, config := range configs {
+func getFilterSection(logInfo serviceLogInfo, logFilters []*logfilter.LogFilter, logFiles *[]string) string {
+	filterSection := ""
+	for _, config := range logInfo.LogConfigs {
 		for _, filterName := range config.Filters {
-			filterValue, ok := filterDefs[filterName]
+			filterValue, ok := findNewestFilter(filterName, logInfo, logFilters)
 			if !ok {
 				plog.WithFields(log.Fields{
-					"service": serviceName,
+					"serviceid": logInfo.ID,
+					"servicename": logInfo.Name,
 					"filter": filterName,
-				}).Warn("log filter not found")
+				}).Warn("service log filter not found")
 				continue
 			}
 			//  CC-3669: do not write duplicate filters for the same log file
 			if !utils.StringInSlice(config.Path, *logFiles) {
-				filters += fmt.Sprintf("\n  if [file] == \"%s\" {\n%s\n  }\n",
+				filterSection += fmt.Sprintf("\n  if [file] == \"%s\" {\n%s\n  }\n",
```

Comment:
That will be sorted out when merging `bespin` back into `develop` (or rebasing `bespin` off of `develop`). Either way there will be some kind of file conflict that will have to be resolved